### ISSUE TITLE
Field Service not enabled in Scratch org even though it is enabled in scratch org definition file

### ISSUE THREAD

**Issue by chfosli** [forcedotcom/cli#918]
### Summary

We have Field Service enabled in scratch org definition file, but yesterday it stopped working. Scratch org is created, but Field Service is not enabled. Other features are working fine.

This breaks a lot of CI/CD scripts, and it prevents us from making new package versions (most critical) since ore metadata has references to WorkOrder and ServiceAppointmen (and more). This objects are not available if Field Service is not enabled.

### Steps To Reproduce:

1.  Create a empty sfdx project
2. Use the following  scratch org definition file: [project-scratch-def.zip](https://github.com/forcedotcom/cli/files/6129375/project-scratch-def.zip)
3. Run `sfdx force:org:create -f config\project-scratch-def.json --setalias test --durationdays 1` to create scatch org

### Expected result

Scratch org is created, and Field Service is enabled under Field Service Settings in Setup

### Actual result

Scratch org is created, but Field Service is NOT enabled under Field Service Settings in Setup.

### Additional information

**SFDX CLI Version** (to find the version of the CLI engine run sfdx --version): 7.85.1

**SFDX plugin Version** (to find the version of the CLI plugin run sfdx plugins --core): 50.16.1

**OS and version**: Windows 10

**Comment by jclark-dot-org** [comment]
We started seeing this issue yesterday (March 11).  According to our pipeline history, this was working as recently as March 8.  Our scratch org definition file includes both
```
"features": [ "FieldService:25", ...]
```
and 
```
    "settings": {
        "fieldServiceSettings": {
            "fieldServiceOrgPref": true
        },
        ...
```
As a workaround - we have an automated build script that:
* creates the scratch
* performs an MDAPI "pre-deploy" of a few items
* installs a few managed packages, including FSL
* `force:source:push`es all source.

I found that adding `FieldService.settings` to the MDAPI pre-deploy does successfully enable FSL, which allows the FSL managed package (and subsequent source push) to succeed.  My `FieldService.settings` is bare-minimum:
```
<?xml version="1.0" encoding="UTF-8"?>
<FieldServiceSettings xmlns="http://soap.sforce.com/2006/04/metadata">
    <fieldServiceOrgPref>true</fieldServiceOrgPref>
</FieldServiceSettings>
```

**Comment by chfosli** [comment]
@jclark-dot-org , your workaround is great for setting up scratch orgs. We are also using unlocked packages to do all deployments to sandbox's and production. When you create a new package version, Salesforce will in the background create a scratch org and push metadata to validate and to run tests (can be skipped when creating package version, but then you can't install it in production). Any ide on who to workaround the bug in that senario?

**Comment by mshanemc** [comment]
Can you try it with the REST workaround to see if that's part of the problem? https://github.com/forcedotcom/cli/issues/942

**Comment by mshanemc** [comment]
There's an investigation happening on this one.

**Comment by chfosli** [comment]
The issue is now fixed and Field Service is now correctly enabled when you create a scratch org.

**Comment by Windyo** [comment]
Hey team, can someone check if this is still fixed ? I notice the same exact behavior on `2.84.6`

Here's my json:

<details>
<summary>Json</summary>
{
  "orgName": "Scratch Org",
  "adminEmail":"REDACTED@REDACTED.REDACTED",
  "country":"FR",
  "language":"en_US",
  "description": "Scratch Org.",
  "hasSampleData": true,
  "release": "current",
  "sourceOrg": "REDACTED",
  "features": [
    "ContactsToMultipleAccounts",
    "DebugApex",
    "EnableSetPasswordInApi",
    "EntityTranslation",
    "Entitlements",
    "ForceComPlatform",
    "Interaction",
    "FlowSites",
    "FieldService:5",
    "FieldServiceDispatcherUser:5",
    "FieldServiceMobileExtension",
    "FieldServiceMobileUser:5",
    "LiveAgent",
    "MarketingUser",
    "MultiCurrency",
    "SalesUser",
    "SalesforceContentUser",
    "StateAndCountryPicklist",
    "RecordTypes",
    "RefreshOnInvalidSession",
    "ServiceCloud",
    "ServiceUser"
  ],
  "settings": {
    "caseSettings": {
      "closeCaseThroughStatusChange" : true
    },
    "CustomAddressFieldSettings": {
     "enableCustomAddressField" : true
      },
    "userManagementSettings": {
        "permsetsInFieldCreation": true,
        "userAccessPoliciesEnabled" : true
      },
    "currencySettings":{ 
        "enableMultiCurrency": true 
      },
    "languageSettings": {
          "enableTranslationWorkbench": true
      },
    "pathAssistantSettings": {
          "pathAssistantEnabled": true
      },
    "lightningExperienceSettings": {
      "enableS1DesktopEnabled": true,
      "enableUsersAreLightningOnly": true,
      "enableLexEndUsersNoSwitching": true
    },
    "mobileSettings": {
      "enableS1EncryptedStoragePref2": false
    },
    "securitySettings": {
      "enableAdminLoginAsAnyUser": true,
      "passwordPolicies": {
        "expiration": "Never",
        "historyRestriction": "0"
      },
      "sessionSettings": {
        "sessionTimeout": "TwelveHours"
      }
    },
    "fieldServiceSettings": {
      "fieldServiceOrgPref": true
    },
   "chatterSettings": {
      "enableChatter": true
    }
  },
  "objectSettings": {
    "product2": {
      "sharingModel": "read"
    },
    "account": {
      "sharingModel": "read"
    },
  	"asset": {
      "sharingModel": "ControlledByParent"
    },
    "contact": {
      "sharingModel": "ControlledByParent"
    }
  }
}


</details>
